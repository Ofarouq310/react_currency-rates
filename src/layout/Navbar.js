import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav>
    <h1 id="navbar-header">
      Currency Rates
    </h1>
    <ul className="nav-list">
      <li>
        <Link to="/" className="nav-link">Home</Link>
      </li>
      <li>
        <Link to="/currencies_list" className="nav-link">Currencies List</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
