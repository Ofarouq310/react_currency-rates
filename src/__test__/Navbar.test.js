import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Navbar from '../layout/Navbar';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/extend-expect';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
