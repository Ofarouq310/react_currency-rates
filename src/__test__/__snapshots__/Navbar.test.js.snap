// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <h1
    id="navbar-header"
  >
    Currency Rates
  </h1>
  <ul
    className="nav-list"
  >
    <li>
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="nav-link"
        href="/currencies_list"
        onClick={[Function]}
      >
        Currencies List
      </a>
    </li>
  </ul>
</nav>
`;
